File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1394.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function findLucky(arr) {
    var map = new Map();
    var result = -1;
    for (var _i = 0, arr_1 = arr; _i < arr_1.length; _i++) {
        var i = arr_1[_i];
        map.set(i, (map.get(i) || 0) + 1);
    }
    for (var _a = 0, _b = map.entries(); _a < _b.length; _a++) {
        var _c = _b[_a], key = _c[0], value = _c[1];
        if (key === value) {
            result = Math.max(result, key);
        }
    }
    return result;
}
;
var arr = [1, 2, 2, 3, 3, 3];
console.log(findLucky(arr));

File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1394.java
import java.util.*;

class Leetcode1394 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        int[] arr = new int[size];
        for(int i = 0; i < size; i++) {
            System.out.printf("Enter %d number element of the array: ", i + 1);
            arr[i] = input.nextInt();
        }
        input.close();
        System.out.println(findLucky(arr));
    }

    private static int findLucky(int[] arr) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int result = -1;
        for(int i : arr) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }
        for(Map.Entry<Integer, Integer> entry: map.entrySet()) {
            if (entry.getKey().equals(entry.getValue())) {
                result = Math.max(result, entry.getKey());
            }
        }
        return result;
    }
}